### PR TITLE
[0.7.0] Attendance

### PR DESCRIPTION
**Pull request by SiravitPhokeed**
<table>
  <tr>
    <td colspan="3"></td>
  </tr>
  <tr>
    <td></td>
    <td></td>
    <td></td>
  </tr>
</table>

**Features**
- Attendance
  - Students and Teachers can view today’s Attendance for their own Classroom
  - Teachers can take Attendance for their own Classroom
- Lookup Classes
- `useConvertContactsForVCard` converts an array of Contacts into a string ready to be used in a vCard

**Fixes**
- `bulkCreateClassrooms` has the `tsx` file extension instead of `ts`
- Fixed all build errors relating to migration to SKCom v3.2.3

**Notes**
Visit https://pr.mysk.school/ to preview this pull request.

### CODE REVIEW

File: utils/helpers/getCurrentSemester.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export default function getCurrentSemester(): 1 | 2 {
+  return 1;
```
Comment (smartwhatt) [inline utils/helpers/getCurrentSemester.ts:2]: That's a problem

**Review by smartwhatt — APPROVED** [review]
image
this is from `/class/student`

also apparently students can take attendance from lookup. and is it a bit weird that if you want to take attendance you can only do it from lookup but not `/class/student`  

other than this I don't find any other problem with the code

**Comment by SiravitPhokeed** [comment]
> this is from /class/student

All non-migrated Lookup pages have this. I left it there because `/class/student` might be overhauled. I’m still thinking.

> also apparently students can take attendance from lookup

No they cannot. They can only view the Attendance data, they just share the same dialog. Again, I have no idea what they actually want from this feature so you’ll just get this half-baked version until I have some direction.

> and is it a bit weird that if you want to take attendance you can only do it from lookup but not /class/student

Again, `/class/student` might be overhauled.

> That's a problem

Yes that is indeed a problem lmao

**Comment by smartwhatt** [comment]
Is it better for the button to show a different message for student, this seem very confusing especially when sometimes students take their own attendance physically
image

**Comment by SiravitPhokeed** [comment]
> Is it better for the button to show a different message for student, this seem very confusing especially when sometimes students take their own attendance physically

I guess that would be a good immediate solution for now. But is that really the implementation we want long-term, or have we not figured that out?

**Comment by smartwhatt** [comment]
The solution is to have an entire page dedicated to these data for teachers and students something like an academic stats (attendance, คะแนนพฤติกรรม, etc) page showing all the data regarding the status of you as a student or student in the class you are advising but in the end, it's not part of this sprint. This sprint is to have a feature ready for next semester to collect data. The ultimate goal for this feature is to have a daily dashboard for the management

**Comment by SiravitPhokeed** [comment]
Changed the verbiage. Can you check both languages and see if they make sense now?

**Comment by smartwhatt** [comment]
LGTM I think

**Comment by SiravitPhokeed** [comment]
Bruh you just commented but didn’t approve

**Review by smartwhatt — APPROVED** [review]
didn't know I have to reapprove

**Pull request merged**